src/Extension/FluentSearchFormExtension.php

```diff
@@ -0,0 +1,93 @@
+ */
+
+namespace TractorCow\Fluent\Extension;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Search\SearchForm;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\SS_List;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Simple extension to enable SS Core search to work with SS Fluent
+ */
+class FluentSearchFormExtension extends SearchForm
+{
+    /**
+     * Return dataObjectSet of the results using current request to get info from form.
+     * Wraps around {@link searchEngine()}.
+     *
+     * @return SS_List
+     */
+    public function getResults()
+    {
+        // Get request data from request handler
+        $request = $this->getRequestHandler()->getRequest();
+
+        $keywords = $request->requestVar('Search');
+
+        $andProcessor = function ($matches) {
+            return ' +' . $matches[2] . ' +' . $matches[4] . ' ';
+        };
+        $notProcessor = function ($matches) {
+            return ' -' . $matches[3];
+        };
+
+        $keyword_patterns = [
+            'andProcessor' => '/()("[^()"]+")( and )("[^"()]+")()/i',
+            'andProcessor' => '/(^| )([^() ]+)( and )([^ ()]+)( |$)/i',
+            'notProcessor' => '/(^| )(not )("[^"()]+")/i',
+            'notProcessor' => '/(^| )(not )([^() ]+)( |$)/i',
+        ];
+
+        foreach ($keyword_patterns as $callback => $pattern) {
+            $keywords = preg_replace_callback($pattern, $$callback, $keywords);
+        }
+
+        $keywords = DB::get_conn()
+            ->escapeString(
+                $this->addStarsToKeywords($keywords)
+            );
+
+        $keywords_strrpl = str_replace(['*', '+', '-'], '', $keywords);
+        $htmlEntityKeywords = htmlentities($keywords, ENT_NOQUOTES, 'UTF-8');
+
+        $current_locale = FluentState::singleton()->getLocale();
+        $sitetree_table_localised = 'SiteTree_Localised_' . $current_locale;
+        $sitetreetables = 'SiteTree_Live.Title, SiteTree_Live.MenuTitle, SiteTree_Live.Content, SiteTree_Live.MetaDescription';
+
+        $query = implode(' ', [
+            'SELECT DISTINCT SiteTree_Live.ID, SiteTree_Live.Title,',
+            "MATCH ({$sitetreetables}) AGAINST ('{$keywords_strrpl}') AS Relevance",
+            'FROM SiteTree_Live',
+            "LEFT JOIN SiteTree_Localised AS {$sitetree_table_localised}",
+            "ON SiteTree_Live.ID = {$sitetree_table_localised}.RecordID AND {$sitetree_table_localised}.Locale = '{$current_locale}' AND (({$sitetree_table_localised}.ID IS NOT NULL))",
+            'WHERE (',
+            "MATCH ({$sitetreetables}) AGAINST ('{$keywords}' IN BOOLEAN MODE)",
+            "+ MATCH ({$sitetreetables}) AGAINST ('{$htmlEntityKeywords}' IN BOOLEAN MODE)",
+            'AND SiteTree_Live.ShowInSearch <> 0',
+            ')',
+            'ORDER BY Relevance DESC',
+            'LIMIT 20'
+        ]);
+
+        $customResults = DB::query($query);
```

Comment:
I highly recommend to use parameterised queries here; There's a high risk of SQL injection with variable substitution in SQL code.

Reply:
Thanks for your suggestion, Damian. Are you meaning all variables in the query need to be parameterised? I'm not an expert with parameterised queries so could be missing basic understand here, but attempting to pass `$sitetreetables` or `$sitetree_table_localised` i.e. table names, through as params, the query fails.

What are your thoughts on changing it to this?
```
$sql = implode(
    ' ',
    [
        'SELECT DISTINCT SiteTree_Live.ID, SiteTree_Live.Title,',
        "MATCH ({$sitetreetables}) AGAINST (?) AS Relevance",
        'FROM SiteTree_Live',
        "LEFT JOIN SiteTree_Localised AS {$sitetree_table_localised}",
        "ON SiteTree_Live.ID = {$sitetree_table_localised}.RecordID AND {$sitetree_table_localised}.Locale = ? AND (({$sitetree_table_localised}.ID IS NOT NULL))",
        "WHERE (MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) + MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) AND SiteTree_Live.ShowInSearch <> 0)",
        'ORDER BY Relevance DESC',
        'LIMIT 20'
    ]
);
$params = [
    $keywords_strrpl,
    $current_locale,
    $keywords,
    $htmlEntityKeywords
];
$custom_results = DB::prepared_query($sql, $params);
```
@tractorcow

Reply:
Easy way to remember;
 - If it's a table or column name, don't parameterise it, but do double quote it
- If it's a value, do parameterise it, use a `?` placeholder.

E.g. 
```php
DB::prepared_query("SELECT * FROM \"{$table\"} WHERE "\"{$table}\".\"{$column}\" = ?", [$value]);
```

Reply:
> $sql = implode(
    ' ',
    [
        'SELECT DISTINCT SiteTree_Live.ID, SiteTree_Live.Title,',
        "MATCH ({$sitetreetables}) AGAINST (?) AS Relevance",
        'FROM SiteTree_Live',
        "LEFT JOIN SiteTree_Localised AS {$sitetree_table_localised}",
        "ON SiteTree_Live.ID = {$sitetree_table_localised}.RecordID AND {$sitetree_table_localised}.Locale = ? AND (({$sitetree_table_localised}.ID IS NOT NULL))",
        "WHERE (MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) + MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) AND SiteTree_Live.ShowInSearch <> 0)",
        'ORDER BY Relevance DESC',
        'LIMIT 20'
    ]
);

Suggest to use HEREDOC for formatting these long-line strings.

```php
$sql = <<<SQL
SELECT DISTINCT "SiteTree_Live"."ID", "SiteTree_Live"."Title"
MATCH ("{$sitetreetables}") AGAINST (?) AS "Relevance"
FROM "SiteTree_Live"
LEFT JOIN "SiteTree_Localised" AS "{$sitetree_table_localised}"
ON
    "SiteTree_Live"."ID" = "{$sitetree_table_localised}"."RecordID"
    AND "{$sitetree_table_localised}"."Locale"" = ?
    AND "{$sitetree_table_localised}"."ID" IS NOT NULL
    WHERE (
        MATCH ("{$sitetreetables}") AGAINST (? IN BOOLEAN MODE) > 0
        AND "SiteTree_Live"."ShowInSearch" > 0
    )
SQL
    ;

Reply:
With heredoc `$` var substitution works, but you don't need to escape double quotes. :)

Reply:
Hey Damian, please have a look now. I've updated the query to be parameterised and have defined the query via HEREDOC.
@tractorcow